password-hash/src/errors.rs

```diff
@@ -20,11 +21,21 @@ pub enum Error {
     /// Cryptographic error.
     Crypto,
 
-    /// Output too short (min 10-bytes).
-    OutputTooShort,
-
-    /// Output too long (max 64-bytes).
-    OutputTooLong,
+    /// Output size unexpected.
+    OutputSize {
+        /// Indicates why the output size is unexpected.
+        ///
+        /// - [`Ordering::Less`]: Size is too small.
+        /// - [`Ordering::Equal`]: Size is not exactly as `expected`.
+        /// - [`Ordering::Greater`]: Size is too long.
+        size: Ordering,
```

Comment:
How about something like `provided` instead of `size`?

Reply:
Great idea!

Reply:
Done.